packages/aws-cdk-lib/aws-eks/lib/aws-auth.ts

```diff
@@ -31,6 +31,13 @@ export class AwsAuth extends Construct {
   constructor(scope: Construct, id: string, props: AwsAuthProps) {
     super(scope, id);
 
+    // Throw if the cluster does not support ConfigMap.
+    const supportConfigMap = props.cluster.authenticationMode !== AuthenticationMode.API ? true : false;
+
+    if (!supportConfigMap) {
+      throw new Error('ConfigMap not supported in the AuthenticationMode');
```

Comment:
nit: can we give better messaging on why `ConfigMap` is not supported?

Reply:
DONE. Just improved the doc string.